app/controllers/report_controller/reports/editor.rb

```diff
@@ -1552,6 +1546,18 @@ def set_form_vars
     end
   end
 
+  def categories_hash
+    cats = Classification.categories.select { |c| c.show }
+    cats.delete_if { |c| c.entries.length == 0 }  # Omit categories for which entries dropdown would be empty.
```

Comment:
suggest combining these two lines:
`cats = Classification.categories.select { |c| c.show && c.entries.size > 0 }`

Maybe in a followup PR, if this is a domain concept (visible categories with entries) we can move this logic to a scope in the model since we have the exact same logic in `get_categories_all` in the chargeback_controller.

I also see something that might be related:
```
app/controllers/miq_policy_controller/alert_profiles.rb:      c if !c.read_only? && c.show && c.entries.size > 0
lib/report_formatter/text.rb:      categories = Classification.categories.collect { |c| c if c.show }.compact
```

The first is visible categories with entries and the second is just visible categories.  It might be worth making these actual things in the model.